pkg/asset/store.go

```diff
@@ -23,16 +32,52 @@ func (s *StoreImpl) Fetch(asset Asset) (*State, error) {
 	return s.fetch(asset, "")
 }
 
+// Load retrieves the state from the state file present in the given directory
+// and returns the assets map
+func (s *StoreImpl) Load(dir string) error {
+	path := filepath.Join(dir, stateFileName)
+	assets := make(map[string]*State)
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
+		return err
+	}
+	err = json.Unmarshal(data, &assets)
+	if err == nil {
+		s.assets = assets
```

Comment:
Hmm, not sure we want to overwrite the in-memory store with the on-disk state file, because we discussed that an asset should always fetch its dependencies to determine whether its dependencies are on disk, if so the asset needs to be regenerated.
https://coreos.slack.com/archives/C68TNFWA2/p1538618620000100

Reply:
cc @staebler ^

Reply:
Agreed. I made a similar argument in my review.